src/socket/tcp.rs

```diff
@@ -1025,9 +1025,18 @@ impl<'a> TcpSocket<'a> {
                 self.timer.set_for_idle(timestamp, self.keep_alive);
             }
 
-            // ACK packets in ESTABLISHED state reset the retransmit timer.
+            // ACK packets in ESTABLISHED state reset the retransmit timer,
+            // except for duplicate ACK packets.
             (State::Established, TcpControl::None) => {
-                self.timer.set_for_idle(timestamp, self.keep_alive);
+                match self.timer {
```

Comment:
This seems obviously correct and OK to merge. Do you think you can extract this part and a dedicated test, and submit a PR for it?

Also, would you mind rewriting the condition a bit? I would add a `Timer::is_retransmit() -> bool` function, and write the condition as `if !timer.is_retransmit() || ack_len != 0 { self.timer.set_for_idle(...) }`